pkg/scheduler/api/hyper_node_info.go

```diff
@@ -638,3 +615,76 @@ func (hni *HyperNodesInfo) nodeMatchRegexSelector(nodeName string, selector topo
 	}
 	return reg.MatchString(nodeName)
 }
+
+// GetAncestors returns all ancestors of a given HyperNode.
+func (hnim HyperNodeInfoMap) GetAncestors(name string) []string {
+	ancestors := []string{name}
+
+	sets.New[string](name)
+	queue := []string{name}
+
+	for len(queue) > 0 {
+		current := queue[0]
+		queue = queue[1:]
+
+		parent := ""
+		hn, ok := hnim[current]
+		if ok && hn.parent != "" {
+			parent = hn.parent
+		} else {
+			parent = hnim.getParent(current)
+		}
+		if ancestorSet := sets.New[string](ancestors...); parent != "" && !ancestorSet.Has(parent) {
+			if !ancestorSet.Has(parent) {
```

Comment:
multi `!ancestorSet.Has(parent)` check?

Reply:
It's copied from old code, I have removed it now